### PR TITLE
Add `accessPolicy` field to Server CRD

### PR DESCRIPTION
**Pull request by alpeb**
Followup to #12844

This new field defines the default policy for Servers, i.e. if a request doesn't match the policy associated to a Server then this policy applies. The values are the same as for `proxy.defaultInboundPolicy` and the `config.linkerd.io/default-inbound-policy` annotation (all-unauthenticated, all-authenticated, cluster-authenticated, cluster-unauthenticated, deny), plus a new value "audit". The default is "deny", thus remaining backwards-compatible.

This field is also exposed as an additional printer column.

### CODE REVIEW

File: charts/linkerd-crds/templates/policy/server.yaml [hunk 1]
```diff
@@ -148,6 +148,11 @@ spec:
                 - required: [podSelector]
                 - required: [externalWorkloadSelector]
               properties:
+                accessPolicy:
```
Comment (mateiidavid) [inline charts/linkerd-crds/templates/policy/server.yaml:151]: Since we are adding a new field to the resource, should we also bump the resource version?

Comment (alpeb) [reply charts/linkerd-crds/templates/policy/server.yaml:151]: I was relying on this being BC to not bump the version, but good call, if someone wants to use this field the only way to guarantee it's supported is through the version.

**Review by mateiidavid — APPROVED** [review]
Looks good to me! With the additional caveat that bumping the version means we once again get into some interesting questions around version skew for multicluster. 

I think we should be fine here though, since a newer client should be able to send requests to an API Server that has an older version of the CRD and that all should work (except the returned resource won't have the new field). We're not reading the field anywhere in the destination service, or service mirror (which require connections to other API Servers). So all in all we shouldn't see any problems here, right?

**Comment by alpeb** [comment]
Thanks Matei, let me summarize what we discussed off-github:
- **Source cluster with newer CRD, target cluster with old CRD**: it breaks! This is the output from the destination controller:
```
level=error msg="Error adding cluster target to store: server CRD (policy.linkerd.io/v1beta3) not found" component=cluster-store
```
The target doesn't know that version exists, so it returns an error.

- **Source cluster with old CRD, target cluster with new CRD**: it works as per my testing. We're still serving both versions and we're using the default "None" conversion strategy, which means that the Server is returned as its stored schema (v1beta3) with only the apiVersion field accommodating to the requested one (v1beta2).
I did some client-go scavenging and confirmed the serializer built in [server_client.go](https://github.com/linkerd/linkerd2/blob/edge-24.7.3/controller/gen/client/clientset/versioned/typed/server/v1beta2/server_client.go#L91) creates a [CodecFactory](https://github.com/kubernetes/apimachinery/blob/v0.30.3/pkg/runtime/serializer/codec_factory.go#L176-L177) leaving the default `CodecFactoryOptions.Strict: false`, so the decoder doesn't complain if there's an unrecognized field.

The first problem disappears as soon as we upgrade the CRDs on the target cluster and restart the destination controller. So something folks need to have clear when upgrading multicluster.

**Pull request merged**